src/python/pants/engine/internals/native_engine.pyi

```diff
@@ -82,6 +82,16 @@ class Snapshot:
     def dirs(self) -> tuple[str, ...]: ...
     @property
     def files(self) -> tuple[str, ...]: ...
+    # Use pants.engine.fs.SnapshotDiff for semantic names
+    def diff(
```

Comment:
Should we just prefix this with an underscore to really ward off callers?

Reply:
Yea, that might make sense. In practice, users will probably consume this via a `@rule` that computes a diff string in some format using difflib...? And so this would be an implementation detail of that `@rule` rather than something consumed directly.

Reply:
The plan is for this to be called in `FmtResult.message` to compute the message. Well not "this" but `pants.engine.fs.SnapshotDiff.from_snapshots`.